guides/index_template.md

```diff
@@ -0,0 +1,189 @@
+# Index Template
+
+Index templates are a convenient way to define settings, mappings, and aliases for one or more indices when they are created. In this guide, you'll learn how to create an index template and apply it to an index.
+
+## Setup
+
+Assuming you have OpenSearch running locally on port 9200, you can create a client instance with the following code:
+
+```rust
+// Let's create a client instance, and an index named `movies`:
+
+// Create a client to make API calls to OpenSearch running on http://localhost:9200.
+
+let url = Url::parse("https://localhost:9200")?;
+let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
+    .cert_validation(CertificateValidation::None)
+    .build()?;
+let client = OpenSearch::new(transport);
```

Comment:
Let's stay consistent with #146 and have the same setup as that, also the first comment refers to creating an index named `movies` which we are not doing, so remove it.
```suggestion
// Create a client to make API calls to OpenSearch running on https://localhost:9200.

let url = Url::parse("https://localhost:9200")?;
let credentials = Credentials::Basic("admin".into(), "admin".into());
let transport = TransportBuilder::new(SingleNodeConnectionPool::new(url))
      .cert_validation(CertificateValidation::None)
      .auth(credentials)
      .build()?;
let client = OpenSearch::new(transport);
```